frontend/src/metabase/dashboard/actions/metadata.js

```diff
@@ -1,16 +1,24 @@
-import { loadMetadataForQueries } from "metabase/redux/metadata";
+import Questions from "metabase/entities/questions";
 import { getMetadata } from "metabase/selectors/metadata";
-
+import { loadMetadataForQueries } from "metabase/redux/metadata";
+import { getLinkTargets } from "metabase/lib/click-behavior";
 import Question from "metabase-lib/Question";
-
 import { isVirtualDashCard } from "../utils";
 
-export const loadMetadataForDashboard = dashCards => (dispatch, getState) => {
+export const loadMetadataForDashboard = dashCards => dispatch => {
+  const cards = dashCards
+    .filter(dc => !isVirtualDashCard(dc)) // exclude text cards
+    .flatMap(dc => [dc.card].concat(dc.series || []));
+
+  dispatch(loadMetadataForCards(cards));
+  dispatch(loadMetadataForLinkedTargets(dashCards));
```

Comment:
We need to load metadata for cards not only that are visible in the dashboard, but also referenced via click actions.